cpp/src/arrow/compute/kernels/scalar_round_arithmetic_test.cc

```diff
@@ -18,7 +18,6 @@
 #include <algorithm>
 #include <cmath>
 #include <memory>
-#include <string>
```

Comment:
Our guideline for includes is [`iwyu`](https://include-what-you-use.org/).  We don't always follow it perfectly (the conformance tool doesn't like type_fwd files) but it is what we aim for.  Please don't remove includes if they are used in the file (I still see many instances of `std::string`) even if the file compiles otherwise (transitive includes are potentially unstable).

Reply:
I've been spoiled by having iwyu being enforced by the compiler.  I wonder what it would take to get the project to a point where we can enjoy that.  For now I'll tell clion to stop warning me about unnecessary includes.